homeassistant/components/switchbot_cloud/light.py

```diff
@@ -0,0 +1,183 @@
+"""Support for the Switchbot Light."""
+
+from typing import Any
+
+from switchbot_api import (
+    CommonCommands,
+    Device,
+    Remote,
+    RGBWWLightCommands,
+    SwitchBotAPI,
+)
+
+from homeassistant.components.light import ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import _LOGGER, HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import SwitchbotCloudData, SwitchBotCoordinator
+from .const import DOMAIN
+from .entity import SwitchBotCloudEntity
+
+
+def value_map_brightness(value: int) -> int:
+    """Return value for brightness map."""
+    return int(value / 255 * 100)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up SwitchBot Cloud entry."""
+    data: SwitchbotCloudData = hass.data[DOMAIN][config.entry_id]
+    async_add_entities(
+        SwitchBotCloudLight(data.api, device, coordinator)
+        for device, coordinator in data.devices.lights
+    )
+
+
+class SwitchBotCloudLight(SwitchBotCloudEntity, LightEntity):
+    """Representation of a SwitchBot Battery Circulator Fan."""
+
+    _attr_max_color_temp_kelvin = 6500
+    _attr_min_color_temp_kelvin = 2700
+
+    _attr_is_on: bool | None = None
+
+    def __init__(
+        self,
+        api: SwitchBotAPI,
+        device: Device | Remote,
+        coordinator: SwitchBotCoordinator,
+    ) -> None:
+        """Entity init."""
+        super().__init__(api, device, coordinator)
+        if device.device_type not in "Strip Light":
+            self._attr_supported_color_modes = {ColorMode.RGB, ColorMode.COLOR_TEMP}
+            _LOGGER.info(
+                "This entity is not Strip Light, May support RGB and COLOR_TEMP mode"
+            )
+        else:
+            _LOGGER.info("This entity is Strip Light, only support RGB mode")
+            self._attr_supported_color_modes = {ColorMode.RGB}
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return True if entity is on."""
+        response: dict | None = self.coordinator.data
+        assert response is not None
+        if self._attr_is_on is None:
+            self._attr_color_mode = ColorMode.RGB
+            power: str | None = response.get("power")
+            self._attr_is_on = "on" in power if power else False
+            self._attr_brightness: int | None = response.get("brightness")
+            attr_rgb_color: str | None = response.get("color")
+
+            self._attr_rgb_color: tuple | None = (
+                tuple(int(i) for i in attr_rgb_color.split(":"))
+                if attr_rgb_color
+                else None
+            )
+            self._attr_color_temp_kelvin: int | None = response.get("colorTemperature")
+        return self._attr_is_on
```

Comment:
a `@property` should only return its state and have no side effects, currently it's setting everything and that is not correct. Instead either overwrite `_handle_coordinator_update` as this is a coordinator entity and use that method to set all the `_attr_*` variables

Reply:
ok